pyproject.toml

```
@@ -58,7 +58,7 @@ dependencies = [
   "numba>=0.57",
   "scipy>=1.15.2; python_version >= '3.10'",
   "scipy>=1.13.1",
-  "z3-solver>=4.12.2",
+  "z3-solver>=4.15.3",
```

```suggestion
  "z3-solver>=4.12.2",
```
Let's revert that change for now. It's unrelated to the PR itself

---

I know which flukes you are speaking about above, but this is different. I just started a [CI run on `main`](https://github.com/munich-quantum-toolkit/qecc/actions/runs/17294024965/job/49087817591), and all `miniums` sessions are failing for the same reason:

```console
File "/home/runner/work/qecc/qecc/src/mqt/qecc/cc_decoder/cli.py", line 14, in <module>
    from . import decoder
  File "/home/runner/work/qecc/qecc/src/mqt/qecc/cc_decoder/decoder.py", line 21, in <module>
    from z3 import Bool, Not, Optimize, Xor, simplify
  File "/home/runner/work/qecc/qecc/.nox/minimums-3-9/lib/python3.9/site-packages/z3/__init__.py", line 1, in <module>
    from .z3 import *
  File "/home/runner/work/qecc/qecc/.nox/minimums-3-9/lib/python3.9/site-packages/z3/z3.py", line 46, in <module>
    from . import z3core
  File "/home/runner/work/qecc/qecc/.nox/minimums-3-9/lib/python3.9/site-packages/z3/z3core.py", line 5, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

I could also reproduce this locally. Only the `minimums` sessions fail; the `tests` sessions are fine. By bumping the version, we can resolve this.

---

Ah. Nevermind then 👍🏼 feel free to resolve and merge